src/components/MobileWalletDialog/routes/CreateWallet/CreateBackupConfirm.tsx

```diff
@@ -21,103 +21,86 @@ import {
+    Array(TEST_COUNT_REQUIRED).fill(null),
   )
 
   const words = useMemo(() => {
     if (!location.state?.vault) return []
     return location.state.vault.getWords() ?? []
   }, [location.state?.vault])
 
-  const randomWordIndices = useMemo(() => {
-    const indices = Array.from({ length: words.length }, (_, i) => i)
-    return indices.sort(() => Math.random() - 0.5).slice(0, 12)
-  }, [words.length])
-
-  const generateTestWords = useCallback((targetWord: string) => {
-    let randomWords = uniq(bip39.generateMnemonic(256).split(' ')) as string[]
-    const otherWords = randomWords
-      .filter(word => word !== targetWord)
-      .sort(() => Math.random() - 0.5)
-      .slice(0, 3)
-
-    const allWords = [...otherWords, targetWord] as string[]
-    return allWords.sort(() => Math.random() - 0.5)
-  }, [])
-
   useEffect(() => {
-    if (selectedWordIndex === null && words.length > 0) {
-      setSelectedWordIndex(0)
-      const targetWord = words[randomWordIndices[0]]
-      setTestWords(generateTestWords(targetWord ?? ''))
+    if (words.length > 0) {
+      setTargetIndices(getRandomIndicesIndexes(words.length, TEST_COUNT_REQUIRED))
     }
-  }, [selectedWordIndex, words, randomWordIndices, generateTestWords])
-
-  const handleWordClick = useCallback(
-    (word: string) => {
-      const currentWordIndex = randomWordIndices[selectedWordIndex ?? 0]
-      if (words[currentWordIndex] === word) {
-        if ((selectedWordIndex ?? 0) + 1 >= TEST_COUNT_REQUIRED) {
-          navigate(MobileWalletDialogRoutes.CreateBackupSuccess, {
-            state: { vault: location.state?.vault },
-          })
-          return
-        }
-
-        setSelectedWordIndex(prev => {
-          const next = (prev ?? -1) + 1
-          const targetWord = words[randomWordIndices[next]]
-          setTestWords(generateTestWords(targetWord ?? ''))
-          return next
-        })
-      } else {
-        const targetWord = words[currentWordIndex]
-        setTestWords(generateTestWords(targetWord ?? ''))
-      }
-    },
-    [
-      randomWordIndices,
-      selectedWordIndex,
-      words,
-      generateTestWords,
-      navigate,
-      location.state?.vault,
-    ],
-  )
+  }, [words])
+
+  const options = useMemo(() => {
+    return targetIndices.map(idx => {
+      const correct = words[idx]
+      const distractors = getRandomWords(bip39.wordlists.english, correct, 2)
+      return uniq([correct, ...distractors]).sort(() => Math.random() - 0.5)
+    })
+  }, [targetIndices, words])
+
+  const isCorrect = useMemo(() => {
+    return selections.every((word, i) => word === words[targetIndices[i]])
+  }, [selections, words, targetIndices])
+
+  const handleSelect = (line: number, word: string) => {
+    setSelections(prev => {
+      const next = [...prev]
+      next[line] = word
+      return next
+    })
+  }
+
+  const hasChosenWords = useMemo(() => {
+    return selections.length >= 3 && selections.every(word => word !== null)
```

Comment:
- see https://github.com/shapeshift/web/pull/9756/files#r2161410254 re: left side always evaluating to true
- use `TEST_COUNT_REQUIRED`

Reply:
Nice catch, done in [5346ecc](https://github.com/shapeshift/web/pull/9756/commits/5346ecc14a9fa14f7a774fde5061e58b867b96a4)